### PR TITLE
Lens support in ROS message generator, examples, tests. 

### PR DESCRIPTION
**Pull request by akru**
## Lens(https://github.com/ekmett/lens) support in message generator

Feature is using lenses as getter/setter of message parts *insead* / together records. 
Records available with prefix `_`.

Example:
Increase Z component of Vector3 by 10:
```haskell
import qualified Ros.Geometry_msgs.Vector3 as V
main = runNode "n" $ subscribe "a" >>= advertise "b" . fmap (V.z +~ 10)
```

And some example from Turtle:
```haskell
import Ros.Turtlesim.Pose
import Ros.Geometry_msgs.Twist
import qualified Ros.Geometry_msgs.Vector3 as V
-- Create Twist message with linear and angular velocity in arguments.
mkTwist :: Real a => a -> a -> Twist
mkTwist l a = def & linear  . V.x .~ realToFrac l
                  & angular . V.z .~ realToFrac a
```

## Constants in message file
Currently constants have a strange names:
```haskell
aCTIVE :: Word.Word8
aCTIVE = 1
```
may be full lower case is more than usual for Haskell constants?
```haskell
active :: Word.Word8
active = 1
```
In this pull constants in messages is full lower case.

### CODE REVIEW

**Comment by acowley** [comment]
This is a really thorough, PR, thank you! One thought I'll throw out there because it's my only hesitation: what about using `lens-family-core` and `lens-family-th` for the message lenses? The advantage of these is that they pull in far fewer dependencies than `lens`.

On the other hand, as build tools like `stack` and Nix gain wider use, having a shared installation of `lens` will become the normal state of affairs. If nobody speaks up, I'll merge the PR as-is shortly.

Thank you again, @akru, I appreciate the care you've clearly taken with the various pieces of code you touched.

**Comment by akru** [comment]
Thank you, `lens-family-core` looks really lite! And generated accessors is compatible with `lens`.
I'll add commit with replacement base lens package to more lightweight `lens-family-core` and `lens-family-th` for auto lens generation in messages.

File: Examples/NodeCompose/NodeCompose.cabal [hunk 1]
```diff
@@ -24,26 +25,28 @@ Executable composition
...
-                   time >= 1.1,
-                   roshask == 0.1.*,
-                   mtl,
-                   ROS-std-msgs,
-                   ROS-sensor-msgs
+  Build-Depends:   base             >= 4.2 && < 6
+                 , vector           >  0.7
+                 , time             >= 1.1
+                 , roshask          >= 0.3
+                 , lens             >= 4.11
```
Comment (acowley) [inline Examples/NodeCompose/NodeCompose.cabal]: This should probably be a `lens-family-core` dependency.

File: Examples/NodeCompose/NodeCompose.cabal [hunk 2]
```diff
@@ -24,26 +25,28 @@ Executable composition
...
```
Comment (acowley) [inline Examples/NodeCompose/NodeCompose.cabal]: And one more dependency to update.

**Comment by acowley** [comment]
Thanks for this! It's an exciting change that should make dealing with nested message types much nicer.

**Pull request merged**